use bullet points for the frontpage

wilzbach opened the pull request:
rebased from #248.

While bullet points itself are already an improvement, we should make sure we put the key features concise there.

Ping @JackStouffer (you are familar with the D's feature overview, right?)

stonemaster commented:
:+1:

JackStouffer commented on public/content/en/welcome/welcome-to-d.md:
```diff
@@ -2,39 +2,31 @@
+- portable code (no undefined behavior)
+- compatible with C application binary interface
+
+... and much more [features](http://dlang.org/overview.html).
```
many more

JackStouffer commented:
LGTM sans comments

JackStouffer commented on public/content/en/welcome/welcome-to-d.md:
```diff
@@ -2,39 +2,31 @@
+- allows memory-safe programming
+- _maintainable_, _easy to understand_ code.
+- evolution of C++ (without the mistakes)
+- short learning curve (similar syntax to C, C++ and Java)
```
To not scare anyone away, I would instead say "(C-like syntax, similar to Java and JavaScript)"

wilzbach replied:
Wouldn't we scare people if we mention Javascript?

stonemaster replied:
I would also rather drop JavaScript. To me it appears that JS doesn't have a really good reputation among software developers. But I like @JackStouffer's suggestion. What about the compromise "(C-like syntax, similar to Java and others)"?

JackStouffer commented on public/content/en/welcome/welcome-to-d.md:
```diff
@@ -2,39 +2,31 @@
+- _maintainable_, _easy to understand_ code.
+- evolution of C++ (without the mistakes)
+- short learning curve (similar syntax to C, C++ and Java)
+- portable code (no undefined behavior)
```
This isn't true. D has UB

ZombineDev replied:
Maybe it should say "less undefined behavior"?

wilzbach replied:
Or should I remove the point entirely?

stonemaster replied:
I am in favour of just dropping that point. "less undefined" may sound confusing

JackStouffer commented on public/content/en/welcome/welcome-to-d.md:
```diff
@@ -2,39 +2,31 @@
+for many diverse languages and has a large number of
+[unique features](http://dlang.org/overview.html):
+
+- _high level language_ with great modeling power
```
D isn't a high level language, maybe instead say "high level constructs for great modeling power"

stonemaster replied:
I like that suggestion.

stonemaster replied:
And: Should we maybe emphasise that D also suits well for functional programming? Something like: "D allows writing code in an imperative, object-oriented or functional style"?

wilzbach commented:
Rebased with feedback incorporated (thanks!) and added these two points:

- multi-paradigm (imperative, structured, object oriented, generic, functional programming purity, and even assembly)
- built-in error prevention (contracts, unittests)

1) How should we sort those points?
2) Should we provide links for these points?

stonemaster commented:
Thanks!

1.) I do like the current sorting. IMO we can just leave it the way it i.
2.) I don't think that we need links for them. We provide already a lot of links throughout the tour and we touch a lot of the topics listed. So links would just visual overhead in my eyes.

wilzbach commented:
> 1.) I do like the current sorting. IMO we can just leave it the way it i.

As I need this for the D-Man on the frontpage (and it's a clear improvement), I will go ahead and merge it.
Please feel free to open a PR to make it even better!

stonemaster commented:
Yep nothing is set in stone :-)
